python/mxnet/gluon/parameter.py

```diff
@@ -904,7 +904,11 @@ def zero_grad(self):
             return
 
         for arr in arrays.values():
-            mx.nd.reset_arrays(*arr, num_arrays=len(arr))
+            if is_np_array():
+                for ele in arr:
+                    ele[:] = 0
+            else:
+                mx.nd.reset_arrays(*arr, num_arrays=len(arr))
```

Comment:
why not always use in-place assign?

Reply:
I’m not sure why we used reset_arrays before. I guess that it would be faster if we use multiple arrays.

Reply:
if that's the case, then we need its equivalence in npx namespace @reminisce

Reply:
Can you add an alias `_npi_reset_arrays` in `reset_arrays.cc`?

Reply:
I've checked the source code. The new approach should be fine as long as we use `cudaMemsetAsync` for implementing `ele[()] = 0`. In fact, `reset_arrays.cc` lies in the `contrib` folder and there is no need to add it to numpy.

Reply:
Sounds good to me.

Reply:
I don't think this is worth having diverging implementation. If reset_arrays is not useful then we should stay away from it.

Reply:
Shall we move away from reset_array in the old ndarary too?

Reply:
I actually do not know why we’ve used the reset_array. This op should be in the contrib while now it’s in the main API. I think this is somehow out-of-the-scope of this PR.